src/core/lib/surface/channel.cc

```diff
@@ -513,8 +513,8 @@ void grpc_channel_destroy(grpc_channel* channel) {
   GRPC_CHANNEL_INTERNAL_UNREF(channel, "channel");
 }
 
-grpc_mdelem grpc_channel_get_reffed_status_elem_slowpath(grpc_channel* channel,
-                                                         int i) {
+grpc_mdelem grpc_channel_get_reffed_status_elem_slowpath(
+    grpc_channel* /*channel*/, int i) {
```

Comment:
This seems like another case where we should consider removing the parameter, although there may actually be a deeper problem here, because it seems strange for a `grpc_channel_*()` method to not actually use the channel for anything... Maybe this method should actually be renamed and possibly moved elsewhere.

Reply:
It seems odd indeed. Looks like it's just a helper for interpreting the status code so it doesn't really have anything to do with the channel

Reply:
This is 4-year old code originally added by @ctiller in #4188. Before that point, this function actually used the `channel` parameter. After #4188, the function had no more explicit reference to the channel but still kept its name and place. It was later split into an inline header and slowpath source component by @arjunroy in #19101 .

Reply:
This is 4-year old code originally added by @ctiller in #4188. Before that point, this function actually used the channel parameter. After #4188, the function had no more explicit reference to the channel but still kept its name and place. It was later split into an inline header and slowpath source component by @arjunroy in #19101 .

Reply:
Found the perfect place to put it: `src/core/lib/transport/status_metadata.{h,cc}` which already exist and turn a metadatum into a status_code . These functions just reverse those.